bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/handler/FreeDeviceHandler.java

```diff
@@ -0,0 +1,145 @@
+    private final Logger logger = LoggerFactory.getLogger(FreeDeviceHandler.class);
+    protected long uptime = -1;
+
+    public FreeDeviceHandler(Thing thing, ZoneId zoneId) {
+        super(thing, zoneId);
+    }
+
+    @Override
+    public void initialize() {
+        super.initialize();
+        try {
+            while (!checkBridgeHandler()) {
+                Thread.sleep(1000);
+            }
+            initializeChannels();
+        } catch (InterruptedException e) {
+            logger.info("Interrupted during initialization", e);
+        }
+    }
+
+    @Override
+    protected void internalPoll() throws FreeboxException {
+        super.internalPoll();
+        fetchSystemConfig();
+    }
+
+    protected abstract DeviceConfig getDeviceConfig() throws FreeboxException;
+
+    protected abstract void internalCallReboot() throws FreeboxException;
+
+    private void initializeChannels() {
+        List<Channel> channels = new ArrayList<>(getThing().getChannels());
+
+        try {
+            DeviceConfig systemConfig = getDeviceConfig();
+            List<Sensor> sensors = systemConfig.getAllSensors();
+            sensors.forEach(sensor -> {
+                ChannelUID sensorId = new ChannelUID(thing.getUID(), PROPERTY_SENSORS, sensor.getId());
+                if (channels.stream().noneMatch(c -> c.getUID().equals(sensorId))) {
+                    ChannelBuilder channelBuilder = ChannelBuilder.create(sensorId).withLabel(sensor.getName());
+                    if (sensor.getKind() == SensorKind.FAN) {
+                        channels.add(channelBuilder.withAcceptedItemType(CoreItemFactory.NUMBER)
+                                .withType(new ChannelTypeUID(BINDING_ID + ":fanspeed")).build());
+                    } else if (sensor.getKind() == SensorKind.TEMP) {
+                        channels.add(channelBuilder.withAcceptedItemType("Number:Temperature")
+                                .withType(new ChannelTypeUID(BINDING_ID + ":temperature")).build());
+                    }
+                }
+            });
+            updateThing(editThing().withChannels(channels).build());
+        } catch (FreeboxException e) {
+            logger.info("Error getting list of optional channels : {}", e.getMessage());
+        }
+    }
+
+    private void fetchSystemConfig() throws FreeboxException {
+        DeviceConfig systemConfig = getDeviceConfig();
+
+        List<Sensor> sensors = systemConfig.getAllSensors();
+        sensors.forEach(sensor -> {
+            switch (sensor.getKind()) {
+                case FAN:
+                    updateChannelDecimal(PROPERTY_SENSORS, sensor.getId(), sensor.getValue());
+                    break;
+                case TEMP:
+                    updateChannelQuantity(PROPERTY_SENSORS, sensor.getId(), sensor.getValue(), SIUnits.CELSIUS);
+                    break;
+                case UNKNOWN:
+                    logger.info("Unknown sensor kind : {}", sensor);
```

Comment:
See above, please review all uses of the logger, thanks.

Reply:
Ok, done.